### DISCUSSION TITLE
xDrip+ integration with Yourloops (Diabeloop)

### DISCUSSION THREAD

**Discussion by fjpezuela** [NightscoutFoundation/xDrip#2156] [Ideas]
There are more and more users using the Yourloops platform (users using the Diabeloop device).

https://www.your-loops.com/

It would be great to be able to integrate with the Yourloops platform and be able to download the data from the platform (bolus, basal, carbohydrates, corrections, etc).

A new hardware data source could be created to download patient data from the platform and be able to upload it to others, such as Nightscout, Tidepool, etc.

Thanks.

**Comment by Navid200** [comment]
@psonnera
@tolot27 Should this become a feature request?

**Comment by tolot27** [comment]
> It would be great to be able to integrate with the Yourloops platform and be able to download the data from the platform (bolus, basal, carbohydrates, corrections, etc).

What would be the benefits of replicating the data? What does xDrip offer which is not a feature of Yourloops and which will never be implemented there? There are so many platforms to collect and visualize data. We should think carefully about which ones to integrate into xDrip and why. There must be a real added value for a larger user base.

Another option is to create a connector to Nightscout.

**Reply by fjpezuela** [reply]
For me, the most important thing about the integration is being able to download the treatments to xDrip and being able to analyze the data on the fly. See what time the bolus was put on, what time he had breakfast, how the graph is going and then be able to upload that data to Nightscout.

The way of visualizing the data in xDrip is very good. It would be something similar to what happens with the download of data from the Medtronic 780 (xDrip Carelink version)

I know many people who have to add insulin and carbohydrates to both diabeloop and xDrip every day, in order to analyze the data and have it uploaded to Nightscout.

**Reply by tolot27** [reply]
> being able to analyze the data on the fly. See what time the bolus was put on, what time he had breakfast, how the graph is going

Isn't this also possible with YourLoops, is it? Just open the website on your phone. BTW: xDrip does not provide any analysis functions. It just show the data. Hence, it would be just a data replication.

Unfortunately, I cannot see the data YourLoops provides because it requires an account. But looking at https://www.dbl-diabetes.de/uberblick-yourloops it looks like they just use a fork of Tidepool.

**Reply by fjpezuela** [reply]
I haven't been able to see YourLoops yet, but it seems to be very similar to the Tidepool platform.

Actually it is what you say, it would be a download of data to xDrip +. That would allow, for example, to be able to pass the COB and IOB data to any watch or bracelet.

What I mean by analyzing the data on the fly is that from xDrip you can better see the times of meals, the times of events (by clicking on the point in question) and analyze, for example, a meal, times of wait etc.

I will have access to Yourloops on June 8. I would be willing to share my data with any of you to see the feasibility of this.

**Comment by fjpezuela** [comment]
Since yesterday I have access to the Yourloops platform.

The integration could also be with Nightscout, as you have mentioned, that way the data could also be sent to xDrip+ or used in AAPS.

If you need anything I can help with, let me know. If you need to see the data or analyze something on the platform.

Thank you very much.
